crates/core/src/context.rs

```diff
@@ -329,6 +346,16 @@ impl Context {
         }
     }
 
+    pub fn spawn_thread<F>(&self, f: F)
```

Comment:
Might be a misleading name as the threads are already spawned and in the pool, so we only execute the task.

Reply:
Hm, good point. So maybe `spawn_task`?